mne/viz/epochs.py

```diff
@@ -442,6 +438,254 @@ def plot_epochs(epochs, epoch_idx=None, picks=None, scalings=None,
+    epochs marked beforehand with rejection parameters.
+
+    Parameters
+    ----------
+
+    epochs : instance of Epochs
+        The epochs object
+    picks : array-like of int | None
+        Channels to be included. If None only good data channels are used.
+        Defaults to None
+    scalings : dict | None
+        Scale factors for the traces. If None, defaults to:
+        `dict(mag=1e-12, grad=4e-11, eeg=20e-6, eog=150e-6, ecg=5e-4, emg=1e-3,
+             ref_meg=1e-12, misc=1e-3, stim=1, resp=1, chpi=1e-4)`
+    n_epochs : int
+        The number of epochs per view. Defaults to 20.
+    n_channels : int
+        The number of channels per view. Defaults to 10.
+    fig_title : str | None
+        The title of the window. If None, epochs name will be displayed.
+        Defaults to None.
+    show : bool
+        Show figure if True. Defaults to True
+    block : bool
+        Whether to halt program execution until the figure is closed.
+        Useful for rejecting bad trials on the fly by clicking on an epoch.
+        Defaults to False.
+
+    Returns
+    -------
+    fig : Instance of matplotlib.figure.Figure
+        The figure.
+
+    Notes
+    -----
+    With trellis set to False, the arrow keys (up/down/left/right) can
+    be used to navigate between channels and time ranges and the
+    scaling can be adjusted with 'page up' and 'page down' keys, but
+    this depends on the backend matplotlib is configured to use
+    (e.g., mpl.use(``TkAgg``) should work).
+    """
+    import matplotlib.pyplot as plt
+    import matplotlib as mpl
+    from matplotlib.collections import LineCollection
+    from matplotlib.colors import colorConverter
+    scalings = _handle_default('scalings_plot_raw', scalings)
+    color = _handle_default('color', None)
+    bad_color = (1.0, 0.0, 0.0)
+
+    epoch_data = epochs.get_data()
+    if picks is None:
+        picks = _handle_picks(epochs)
+    if len(picks) < 1:
+        raise RuntimeError('No appropriate channels found. Please'
+                           ' check your picks')
+    n_epochs = min(n_epochs, len(epochs.events))
+    duration = len(epochs.times) * n_epochs
+    n_channels = min(n_channels, len(picks))
+    times = epochs.times * 1e3
+
+    # Reorganize channels
+    inds = list()
+    types = list()
+    for t in ['grad', 'mag']:
+        idxs = pick_types(epochs.info, meg=t, ref_meg=False, exclude=[])
+        if len(idxs) < 1:
+            continue
+        mask = np.in1d(idxs, picks, assume_unique=True)
+        inds.append(idxs[mask])
+        types += [t] * len(inds[-1])
+    pick_kwargs = dict(meg=False, ref_meg=False, exclude=[])
+    for t in ['eeg', 'eog', 'ecg', 'emg', 'ref_meg', 'stim', 'resp',
+              'misc', 'chpi', 'syst', 'ias', 'exci']:
+        pick_kwargs[t] = True
+        idxs = pick_types(epochs.info, **pick_kwargs)
+        if len(idxs) < 1:
+            continue
+        mask = np.in1d(idxs, picks, assume_unique=True)
```

Comment:
Yeah, idxs is a list of ints as is picks. It is comparing them to find out the correct indices, since picks are not necessarily the same.